byteps/mxnet/__init__.py

```diff
@@ -221,8 +221,6 @@ def _init_params(self):
                 param_arrays = param._check_and_get(param._data, list)
                 idx = self._param2idx[param.name]
 
-                if rank() != self.root_rank:
-                    param_arrays[0].__imul__(0)
                 byteps_push_pull(param_arrays[0], version=0, priority=0,
                                  name="parameter_" + str(idx), is_average=False)
```

Comment:
May be we should use `is_average=True` here?

Reply:
I think there is no need to average, since the server just selects one received tensor and copy it to the stored buffer.

Reply:
You're right.